### PR TITLE
Improve redis image readiness checks

### PR DESCRIPTION
**Pull request by willdollman**
- [x] Check that `redis-cli --raw incr ping` command works in container image tests
- [x] Add healthcheck script to image as suggested in https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/419

Readiness check testing:

```
$ /usr/local/bin/readiness.sh 
Could not connect to Redis at 127.0.0.1:6379: Connection refused
/redis-data $ echo $?
1

# Start redis
redis-server &

/redis-data $ /usr/local/bin/readiness.sh 
/redis-data $ echo $?
0
```

## Test plan

- CI tests
- Tested readiness check locally

### CODE REVIEW

File: docker-images/redis-cache/BUILD.bazel [hunk 1]
```diff
@@ -10,12 +10,17 @@ filegroup(
 
 pkg_tar(
     name = "redis_tarball",
-    srcs = [":redis_conf"],
+    srcs = [
+        ":redis_conf",
+        "//docker-images/redis-store:redis_readiness",
```
Comment (eseliger) [inline docker-images/redis-cache/BUILD.bazel:15]: Nitpick:

not sure if it matters, but should we either duplicate this file, or store it outside of redis-store, so we don't "reach across packages" here?

**Review by eseliger — APPROVED** [review]
Screenshot 2024-02-09 at 13 25 35@2x

Looks like he will!

Great work, thanks for the quick followup :)

**Comment by daxmc99** [comment]
@willdollman Is this in the latest image? 
`docker run -it --entrypoint sh sourcegraph/redis-cache:262142_2024-02-13_5.2-4db83b41b419`

```
/ $ ls -al
total 60
drwxr-xr-x    1 root     root          4096 Feb 13 17:42 .
drwxr-xr-x    1 root     root          4096 Feb 13 17:42 ..
-rwxr-xr-x    1 root     root             0 Feb 13 17:42 .dockerenv
drwxr-xr-x    2 root     root          4096 Feb  9 03:05 bin
drwxr-xr-x    5 root     root           360 Feb 13 17:42 dev
drwxr-xr-x    1 root     root          4096 Feb 13 17:42 etc
drwxr-xr-x    1 root     root          4096 Feb  9 03:05 home
drwxr-xr-x    3 root     root          4096 Feb  9 03:05 lib
lrwxrwxrwx    1 root     root             3 Feb  9 03:05 lib64 -> lib
drwxr-xr-x    2 root     root          4096 Feb  9 03:05 opt
dr-xr-xr-x  366 root     root             0 Feb 13 17:42 proc
drwxr-xr-x    2 redis    redis         4096 Feb  9 03:05 redis-data
drwx------    2 root     root          4096 Feb  9 03:05 root
drwxr-xr-x    2 root     root          4096 Feb  9 03:05 run
drwxr-xr-x    2 root     root          4096 Feb  9 03:05 sbin
dr-xr-xr-x   11 root     root             0 Feb 13 17:42 sys
drwxrwxrwt    2 root     root          4096 Feb  9 03:05 tmp
drwxr-xr-x    1 root     root          4096 Jan  1  2000 usr
drwxr-xr-x    8 root     root          4096 Feb  9 03:05 var
```

**Comment by eseliger** [comment]
oh, seems like `/readiness.sh` was a mistake on my end - seems like it's remapped to `"/readiness.sh": "/usr/local/bin/readiness.sh"`

**Comment by willdollman** [comment]
It should be @daxmc99 , the change is in the Bazel part so doesn't need a base image update

```
~> docker run -it --entrypoint sh sourcegraph/redis-cache:262142_2024-02-13_5.2-4db83b41b419
Unable to find image 'sourcegraph/redis-cache:262142_2024-02-13_5.2-4db83b41b419' locally
262142_2024-02-13_5.2-4db83b41b419: Pulling from sourcegraph/redis-cache
58f2dc3e4d84: Pull complete
9eebacebe2d4: Pull complete
Digest: sha256:cbcf96a575268b2011c7f87a7fbfc9f27bdc34391254aa904dc3c7a3b4361e31
Status: Downloaded newer image for sourcegraph/redis-cache:262142_2024-02-13_5.2-4db83b41b419
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
/redis-data $ ls -al /usr/local/bin/
total 4
drwxr-xr-x    1 root     root            24 Jan  1  2000 .
drwxr-xr-x    1 root     root             6 Jan  1  2000 ..
-rwxr-xr-x    1 root     root           109 Jan  1  2000 readiness.sh
```

**Pull request merged**